### PR TITLE
HV-1072 Fix for cascaded method parameter validation in combination with group sequence

### PR DESCRIPTION
**Pull request by jwillebrands**

### CODE REVIEW

**Comment by gsmet** [comment]
@jwillebrands will review, thanks!

Could you sign the CLA for Hibernate Validator here: https://cla.jboss.org/ ?

**Comment by gunnarmorling** [comment]
@jwillebrands, could you change the author email of the commits to the address you are using with your GitHub account? That way the changes will properly be attributed to your user in the history and stats. Thanks!

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java [hunk 1]
```diff
@@ -1151,12 +1150,10 @@ private int validatePropertyForNonDefaultGroup(ValueContext<?, Object> valueCont
 		while ( sequenceIterator.hasNext() ) {
 			Sequence sequence = sequenceIterator.next();
 			for ( GroupWithInheritance groupOfGroups : sequence ) {
-				int numberOfFailingConstraint = 0;
+				int numberOfViolations = validationContext.getFailingConstraints().size();
 
 				for ( Group group : groupOfGroups ) {
-					numberOfFailingConstraint += validateParametersForGroup(
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java]: Could you apply the same fix for return value validation?

**Comment by gunnarmorling** [comment]
Rebased and applied. Many thanks, @jwillebrands; great fix and very thorough testing!

**Comment by gunnarmorling** [comment]
Added one more commit to do some clean-up regarding whitespace (we are using tabs traditionally, and had introduced a Checkstyle rule for that in parallel to your work).

**Pull request closed without merging**